pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-common/src/main/java/org/apache/pinot/plugin/ingestion/batch/common/BaseSegmentPushJobRunner.java

```diff
@@ -0,0 +1,147 @@
+import org.apache.pinot.spi.ingestion.batch.spec.SegmentGenerationJobSpec;
+
+public abstract class BaseSegmentPushJobRunner implements IngestionJobRunner {
+
+  protected SegmentGenerationJobSpec _spec;
+  protected String[] _files;
+  protected PinotFS _outputDirFS;
+  protected URI _outputDirURI;
+  protected TableConfig _tableConfig;
+  protected boolean _consistentPushEnabled;
+
+  /**
+   * Initialize BaseSegmentPushJobRunner with SegmentGenerationJobSpec
+   * Checks for required parameters in the spec and enablement of consistent data push.
+   */
+  @Override
+  public void init(SegmentGenerationJobSpec spec) {
+    _spec = spec;
+    if (_spec.getPushJobSpec() == null) {
+      throw new RuntimeException("Missing PushJobSpec");
+    }
+
+    // Read Table spec
+    if (_spec.getTableSpec() == null) {
+      throw new RuntimeException("Missing tableSpec");
+    }
+
+    // Read Table config
+    if (_spec.getTableSpec().getTableConfigURI() == null) {
+      throw new RuntimeException("Missing property 'tableConfigURI' in 'tableSpec'");
+    }
+
+    _tableConfig = SegmentGenerationUtils.getTableConfig(_spec.getTableSpec().getTableConfigURI(), spec.getAuthToken());
+    _consistentPushEnabled = ConsistentDataPushUtils.consistentDataPushEnabled(_tableConfig);
+  }
+
+  /**
+   * Initialize filesystems and obtain the raw input files for upload.
+   */
+  public void initFileSys() {
+    // Init all file systems
+    List<PinotFSSpec> pinotFSSpecs = _spec.getPinotFSSpecs();
+    for (PinotFSSpec pinotFSSpec : pinotFSSpecs) {
+      PinotFSFactory.register(pinotFSSpec.getScheme(), pinotFSSpec.getClassName(), new PinotConfiguration(pinotFSSpec));
+    }
+
+    // Get outputFS for writing output Pinot segments
+    try {
+      _outputDirURI = new URI(_spec.getOutputDirURI());
+      if (_outputDirURI.getScheme() == null) {
+        _outputDirURI = new File(_spec.getOutputDirURI()).toURI();
+      }
+    } catch (URISyntaxException e) {
+      throw new RuntimeException("outputDirURI is not valid - '" + _spec.getOutputDirURI() + "'");
+    }
+    _outputDirFS = PinotFSFactory.create(_outputDirURI.getScheme());
+
+    // Get list of files to process
+    try {
+      _files = _outputDirFS.listFiles(_outputDirURI, true);
+    } catch (IOException e) {
+      throw new RuntimeException("Unable to list all files under outputDirURI - '" + _outputDirURI + "'");
+    }
+  }
+
+  /**
+   * Returns segment names, which will be supplied to the segment replacement protocol as the new set of segments to
+   * atomically update when consistent data push is enabled.
+   * @param segmentsUriToTarPathMap Map from segment URI to corresponding tar path. Either the URIs (keys), the
+   *                                tarPaths (values), or both may be used depending on upload mode.
+   */
+  public List<String> getSegmentsToReplace(Map<String, String> segmentsUriToTarPathMap) {
+    return SegmentPushUtils.getSegmentNames(segmentsUriToTarPathMap.values());
```

Comment:
Let's put the logic here instead of putting in `Util` class. We usually extract the logic to `Util` class when the same code needs to be called in multiple places to avoid the duplicate code. In this case, we only need to use this function here.